imnodes.cpp

```diff
@@ -2398,6 +2398,24 @@ void SetNodeDraggable(int node_id, const bool draggable)
     node.draggable = draggable;
 }
 
+ImVec2 GetNodeGridSpacePos(int node_id)
+{
+    EditorContext& editor = editor_context_get();
+    const int node_idx = object_pool_find(editor.nodes, node_id);
+    assert(node_idx != -1);
+    NodeData& node = editor.nodes.pool[node_idx];
+    return node.origin;
+}
+
+ImVec2 GetNodeScreenSpacePos(const int node_id)
+{
+    EditorContext& editor = editor_context_get();
+    const int node_idx = object_pool_find(editor.nodes, node_id);
+    assert(node_idx != -1);
+    NodeData& node = editor.nodes.pool[node_idx];
+    return grid_space_to_editor_space(node.origin);
```

Comment:
If you really want the node's coordinates in the window's screen space coordinates, we need to use a different transform, something called `grid_space_to_screen_space`. The code doesn't yet contain such a function, but it's the exact opposite of `screen_space_to_grid_space`:

```cpp
// something like this
inline ImVec2 grid_space_to_screen_space(const ImVec2& v)
{
    const EditorContext& editor = editor_context_get();
    return v + g.canvas_origin_screen_space + editor.panning;
}
```

I did this really quick and beautiful drawing to try to better explain the coordinate systems that are in imnodes 😅 What you've done is almost correct, but we still need to add the offset of the node editor window's coordinates to the coordinate you've calculated.

<img width="650" alt="Screenshot 2020-10-10 at 13 37 29" src="https://user-images.githubusercontent.com/3276684/95653052-49240380-0afe-11eb-8a2f-85d8fb384a3f.png">